### PR TITLE
Support for older anaconda boot line options

### PR DESCRIPTION
**Pull request by xavierba**
Prior to RHEL 7 (and Fedora 17), anaconda was using different boot line options.
Cobbler has been (partially) switched to the newer options with https://github.com/cobbler/cobbler/commit/902d0c43b861f4eae360861acf1f07ced74a0648. However, while this allows cobbler to work with anaconda versions that have removed the deprecated options (like CentOS Stream 9), this breaks anaconda versions that don't have the newer options, that is everything earlier than RHEL 7 and Fedora 17. All these distributions are indeed not supported anymore, but there are still use cases where installing these legacy distros is needed.

This PR adds a 'redhatlegacy' breed, which supports the older anaconda options.
It also converts the legacy 'kssendmac' to 'inst.ks.sendmac', where applicable.

### CODE REVIEW

**Comment by olafhering** [comment]
What this patch declares as `legacy` is seen as legacy today. But there will be the time when today's variant will be considered legacy as well.

I think a better, future-proof naming scheme should be used.

File: cobbler/tftpgen.py [hunk 1]
```diff
@@ -864,10 +864,14 @@ def build_kernel_options(self, system, profile, distro, image, arch: str, autoin
                     autoinstall_path = "http://%s/cblr/svc/op/autoinstall/profile/%s" \
                                        % (httpserveraddress, profile.name)
 
-            if distro.breed is None or distro.breed == "redhat":
+            if distro.breed is None or distro.breed == "redhat" or distro.breed == "redhatlegacy":
```
Comment (vzhestkov) [inline cobbler/tftpgen.py:867]: Maybe do it this way?
```suggestion
            if distro.breed in [None, "redhat", "redhatlegacy"]:
```

File: cobbler/actions/buildiso.py [hunk 2]
```diff
@@ -217,10 +217,13 @@ def generate_netboot_iso(self, imagesdir, isolinuxdir, profiles=None, systems=No
                 else:
                     append_line += " autoyast=%s" % data["autoinstall"]
 
-            if dist.breed == "redhat":
+            if dist.breed in ["redhat", "redhatlegacy"]:
```
Comment (vzhestkov) [inline cobbler/actions/buildiso.py:220]: Maybe get rid of separate `if`s and switch to `if ... elif ...`? Or maybe fix it in the separate PR.

File: cobbler/tftpgen.py [hunk 3]
```diff
@@ -864,10 +864,14 @@ def build_kernel_options(self, system, profile, distro, image, arch: str, autoin
...
 
-                append_line += " kssendmac"
-                append_line = "%s inst.ks=%s" % (append_line, autoinstall_path)
+                if distro.breed == "redhatlegacy":
+                    append_line += " kssendmac"
+                    append_line = "%s ks=%s" % (append_line, autoinstall_path)
+                else:
+                    append_line += " inst.ks.sendmac"
+                    append_line = "%s inst.ks=%s" % (append_line, autoinstall_path)
```
Comment (vzhestkov) [inline cobbler/tftpgen.py:874]: Quite strange mix of `append_line += ...` and `append_line = "%s ... %s" % (append_line, ...)`
```suggestion
                if distro.breed == "redhatlegacy":
                    append_line += " kssendmac ks=%s" % autoinstall_path
                else:
                    append_line += " inst.ks.sendmac inst.ks=%s" % autoinstall_path
```

**Review by vzhestkov — CHANGES_REQUESTED** [review]

File: cobbler/tftpgen.py (hunk 3 above)
Comment (xavierba) [reply cobbler/tftpgen.py:874]: The mix is what the code was doing before I touched it, so I left it this way and changed only kssendmac/inst.ks.sendmac, but that can indeed be made into one line.

File: cobbler/tftpgen.py (hunk 1 above)
Comment (xavierba) [reply cobbler/tftpgen.py:867]: yes, that is more readable

File: cobbler/actions/buildiso.py (hunk 2 above)
Comment (xavierba) [reply cobbler/actions/buildiso.py:220]: I would think refactoring work doesn't really belong to this patch, so I'd rather keep this for a separate PR.

**Comment by xavierba** [comment]
@olafhering agreed, legacy is purely a matter of which point in time you look at it, but I couldn't think of anything better. I'm definitely open to suggestion.

**Comment by xavierba** [comment]
@vzhestkov thanks for looking at the PR. Hopefully I've addressed your concerns now.

**Review by vzhestkov — APPROVED** [review]
@xavierba thanks, approved

**Review by SchoolGuy — CHANGES_REQUESTED** [review]
This PR is conflicting with #2416 sadly. I will personally rebase this PR after we merged my refactoring. I will also then think of a better naming schema. Thanks for bringing this up!

**Comment by olafhering** [comment]
> @olafhering agreed, legacy is purely a matter of which point in time you look at it, but I couldn't think of anything better. I'm definitely open to suggestion.

Maybe something like `pre_rhel_N` or `pre_fedora_N` will make it more obvious.

**Comment by xavierba** [comment]
@SchoolGuy Is there anything I can do to help with this ?

**Comment by SchoolGuy** [comment]
@xavierba Nope. We are in the process of rebasing and fixing up the review of above mentioned PR. I havn't forgotten this, just busy a lot fixing all the things which need fixing. If all goes will, this PR goes into `master` this week.

**Comment by heroin-moose** [comment]
Maybe just a version check? We already know which versions of rhel need `ks` and which need `inst.ks`.

**Comment by xavierba** [comment]
That one and only commit should hopefully address concerns raised in this thread. No new breed creation, and thus no naming issue, no moving around in the signature file and no change to tests.
This is all replaced by a check on the os_version, as suggested by @heroin-moose.
Also, this is on top of current master, where the previously conflicting refactoring has already landed.
However, none of this has been tested yet, so I'm not updating the PR yet.
Would you mind taking a look and confirm that's what you wanted ?
https://github.com/xavierba/cobbler/commit/8f5dcfffadfc83b0e2052e5c5d0afc2a5d64aa5d

**Comment by olafhering** [comment]
I approve the minimal non-instrusive change 8f5dcfffadfc83b0e2052e5c5d0afc2a5d64aa5d

**Comment by xavierba** [comment]
Thanks @olafhering, I've opened a new PR #2888.

**Comment by xavierba** [comment]
This was replaced #2888.

**Pull request closed without merging**